### PR TITLE
Feature.imu

### PR DESCRIPTION
**Pull request by bjornborg**
Initial release of the proxy software interface for the altimu from pololu.

Altimeter is extremely inaccurate since it based from pressure measurements (standard formulae assumptions).

Calibrations in accelerometers, gyroscope and magnetometer (soft iron calibration) are left to do.
Partial calibration (basic) in magnetometer is implemented and saved externally in a file which is carried over to next execution.

Todo: Initialisation of the tilting in the imu for the acceleration i.e. compensate the initial pitch and roll (under assumption the the vehicle is on a flat plane).

Possible todo: normalise the magnetic field vector since it most likely do not carry any additional information of value to us.

Renamed some of the IMU odvd message fields to more proper names.

### CODE REVIEW

File: code/core/system/proxy-imu/src/Pololualtimu10device.cpp [hunk 1]
```diff
@@ -0,0 +1,524 @@
...
+namespace proxy {
+
+/**
+ * Constructor for PololuAltImuV5 device interfacing through I2C.
+ *
+ */
+PololuAltImu10Device::PololuAltImu10Device(std::string const &a_deviceName, std::string &a_calibrationFile, bool &a_debug)
+    : m_deviceFile()
+    , m_calibrationFile(a_calibrationFile)
+    , m_calibrationPath("/opt/opendlv.core.configuration/")
```
Comment (olbender) [inline code/core/system/proxy-imu/src/Pololualtimu10device.cpp]: We should avoid absolute paths.

File: code/core/system/proxy-imu/include/Pololualtimu10device.h [hunk 2]
```diff
@@ -0,0 +1,233 @@
...
+    float m_compassMinVal[3];
+
+    float m_heavyAcc[3];
+
+    bool m_initialized;
+
+    bool m_debug;
+};
+
+namespace lis3RegAddr {
```
Comment (olbender) [inline code/core/system/proxy-imu/include/Pololualtimu10device.h:93]: I don't think we should use namespaces like this, instead put the name in the enum name. Also make sure to use an informative name.

File: code/core/system/proxy-imu/src/ProxyIMU.cpp [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+    , m_debug() {
+}
+
+ProxyIMU::~ProxyIMU() {
+}
+
+odcore::data::dmcp::ModuleExitCodeMessage::ModuleExitCode ProxyIMU::body() {
+    while (getModuleStateAndWaitForRemainingTimeInTimeslice() == odcore::data::dmcp::ModuleStateMessage::RUNNING) {
+
+        auto gyroscopeReading = m_device->ReadGyroscope();
```
Comment (olbender) [inline code/core/system/proxy-imu/src/ProxyIMU.cpp:52]: The proper messages should be used here.

**Review by olbender — CHANGES_REQUESTED** [review]
Please address the comments. Also we are missing test cases, which we really should have in opendlv.core.

File: code/core/system/proxy-imu/src/Pololualtimu10device.cpp (hunk 1 above)
Comment (bjornborg) [reply code/core/system/proxy-imu/src/Pololualtimu10device.cpp]: Fixed

File: code/core/system/proxy-imu/include/Pololualtimu10device.h (hunk 2 above)
Comment (bjornborg) [reply code/core/system/proxy-imu/include/Pololualtimu10device.h:93]: To be discussed further

File: code/core/system/proxy-imu/src/ProxyIMU.cpp (hunk 3 above)
Comment (bjornborg) [reply code/core/system/proxy-imu/src/ProxyIMU.cpp:52]: Proper messages are used, in "auto".

**Review by olbender — APPROVED** [review]
Good. But we need test cases later.

**Pull request merged**